File: depscanner/test/customer-app-fixtures/gem-sinatra-erb-injection/safe/views/renderer.rb
require 'erb'
require 'cgi'

# Patched: ERB source is a hard-coded literal compiled at class-load time.
# The user-supplied value is HTML-escaped via CGI.escapeHTML (a registered
# xss sanitizer in sinatra.yaml) before being interpolated as data.
class Renderer
  FIXED_TEMPLATE = ERB.new('<p>Hello, <%= @name %></p>')

  def self.render_greeting(name)
    safe_name = CGI.escapeHTML(name)
    @name = safe_name
    FIXED_TEMPLATE.result(binding)
  end
end

File: depscanner/test/customer-app-fixtures/pypi-flask-jinja-attr/safe/services/renderer.py
import jinja2
import html

# Patched: the Jinja template source is a hard-coded literal compiled at
# module load. The user-supplied value is HTML-escaped via the stdlib
# `html.escape(*)` sanitizer (registered in flask.yaml) before it reaches
# Template.render, so the bundled `*.render(*)` xss sink sees no live taint.
_FIXED_TEMPLATE = jinja2.Template('<p>Hello, {{ name }}</p>', autoescape=True)


def render_user_greeting(name):
    safe_name = html.escape(name)
    return _FIXED_TEMPLATE.render(name=safe_name)
